### PR TITLE
Provide CMake exported targets

### PR DESCRIPTION
**Pull request by mwoehlke-kitware**
Modify build/install rules to provide a CMake package configuration file with modern exported targets. This makes it much easier for users to use FCL:

    find_package(fcl)
    target_link_libraries(myapp fcl) # takes care of includes too!

I wrote this in a way that the new logic only takes effect when compiled with a more recent version of CMake, because I wasn't sure of the impact of bumping the minimum required CMake version. If it's okay with the project to require 2.8.12, please let me know and I can simplify the logic accordingly.

### CODE REVIEW

**Comment by sherm1** [comment]
Thanks, Matthew! My preference would be to require CMake >= 2.8.12 and simplify the logic. Does anyone object to that?

**Comment by sherm1** [comment]
If I'm reading [this](http://packages.ubuntu.com/trusty/cmake) correctly the default CMake for Ubuntu 14.04 is 2.8.12.

**Comment by sherm1** [comment]
Linux CI builds failed.

**Comment by mwoehlke-kitware** [comment]
> My preference would be to require CMake >= 2.8.12 and simplify the logic.

Done, as a separate commit so I can easily back it out again if there are objections. Hopefully the CI is also fixed (I don't have octomap in my build which is why it was working on my machine).

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -5,14 +5,31 @@ else()
...
+  FILE "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake")
 
 install(TARGETS ${PROJECT_NAME}
+  EXPORT ${PROJECT_NAME}Config
   ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
   LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
   RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
+install(EXPORT ${PROJECT_NAME}Config
+  DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
```
Comment (traversaro) [inline src/CMakeLists.txt]: With this installation location, `fclConfig.cmake` will not be found automatically on Windows even if the system `CMAKE_PREFIX_PATH` enviromental variable matches the `CMAKE_INSTALL_PREFIX` of `fcl`. [1]
Modifying the code as in [2] :
~~~
if(WIN32 AND NOT CYGWIN)
  set(_EXPORT_INSTALL_DESTINATION CMake)
else()
  set(_EXPORT_INSTALL_DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME})
endif()

install(EXPORT ${PROJECT_NAME}Config
        DESTINATION ${_EXPORT_INSTALL_DESTINATION})
~~~ 
will ensure that the library will be easily found also in Windows.

[1] : https://cmake.org/cmake/help/v3.0/command/find_package.html
[2] : https://github.com/robotology/ycm/blob/master/modules/InstallBasicPackageFiles.cmake#L176

**Comment by mwoehlke-kitware** [comment]
Ping? CI passes now. Is this just waiting on a suitable period for objections to elapse, or is there something else I should change?

**Comment by sherm1** [comment]
> Ping? CI passes now. 

Sorry, GitHub doesn't notify when new commits are pushed so I don't think anyone noticed this is ready. A ping definitely helps! 

Thanks, Matthew!

**Comment by mwoehlke-kitware** [comment]
Thanks, @sherm1!

**Pull request merged**